ocs_ci/ocs/node.py

```diff
@@ -908,6 +914,106 @@ def delete_and_create_osd_node_vsphere_upi(osd_node_name, use_existing_node=Fals
     return new_node_name
 
 
+def delete_and_create_osd_node_vsphere_lso(osd_node_name, use_existing_node=False):
+    """
+    Unschedule, drain and delete osd node, and creating a new osd node.
+    At the end of the function there should be the same number of osd nodes as
+    it was in the beginning, and also ceph health should be OK.
+    This function is for vSphere UPI.
+
+    Args:
+        osd_node_name (str): the name of the osd node
+        use_existing_node (bool): If False, create a new node and label it.
+            If True, use an existing node to replace the deleted node
+            and label it.
+
+    Returns:
+        str: The new node name
+
+    """
+    from ocs_ci.ocs.platform_nodes import PlatformNodesFactory
+    from ocs_ci.ocs.resources.storage_cluster import get_osd_size
+
+    sc_name = constants.LOCAL_BLOCK_RESOURCE
+    old_pv_objs = get_pv_objs_in_sc(sc_name)
+
+    osd_node = get_node_objs(node_names=[osd_node_name])[0]
+    osd_pod = get_node_pods(osd_node_name, pods_to_search=pod.get_osd_pods())[0]
+    osd_id = pod.get_osd_pod_id(osd_pod)
+    log.info(f"osd id to remove = {osd_id}")
+
+    log.info("Scale down node deployments...")
+    scale_down_deployments(osd_node_name)
+    log.info("Scale down deployments finished successfully")
+    remove_nodes([osd_node])
+    log.info(f"name of deleted node = {osd_node_name}")
+
+    if config.ENV_DATA.get("rhel_workers"):
+        node_type = constants.RHEL_OS
+    else:
+        node_type = constants.RHCOS
+
+    if not use_existing_node:
+        log.info("Preparing to create a new node...")
+        new_node_names = add_new_node_and_label_upi(node_type, 1)
+        new_node_name = new_node_names[0]
+    else:
+        node_not_in_ocs = get_worker_nodes_not_in_ocs()[0]
+        log.info(
+            f"Preparing to replace the node {osd_node_name} "
+            f"with an existing node {node_not_in_ocs.name}"
+        )
+        if node_type == constants.RHEL_OS:
+            set_selinux_permissions(workers=[node_not_in_ocs])
+        label_nodes([node_not_in_ocs])
+        new_node_name = node_not_in_ocs.name
+
```

Comment:
This code can me replaced with `delete_and_create_osd_node_vsphere_upi` because it's doing same task

Reply:
Yes, you right. Generally, I created this issue https://github.com/red-hat-storage/ocs-ci/issues/3648 so that we have a function for every platform and deployment, and we don't need to check which function to use every time.

Reply:
Fixed